src/redund.jl

```diff
@@ -38,16 +52,36 @@ function checkredundancy{T}(A::AbstractMatrix{T}, b::AbstractVector{T},
 end
 
 
-"""Check if `λ` is a line of matrix `A`. `λ` might not have the same `eltype` as `A` and `b` as it might have been scaled by `normalizecut`."""
-function isinside{T}(A::AbstractMatrix{T}, b::AbstractVector{T}, λ::AbstractVector, isfun::Bool, tol::Float64)
+"""Check if `λ` is a line of matrix `A`. `λ` might not have the same `eltype`
+as `A` and `b` as it might have been scaled by `normalizecut`."""
+function isinside{T}(A::AbstractMatrix{T}, b::AbstractVector{T},
+                     λ::AbstractVector, isfun::Bool, tol::Float64)
     nlines = size(A, 1)
 
     check = false
     k = 0
     while ~check && k < nlines
         k += 1
         a, β = normalizedcut(A, b, k, isfun, tol)
-        check = norm(a - λ, Inf) < tol
+        check = (norm(a - λ, Inf) < tol)
     end
     check, k
 end
+
+
+checkredundancy(A, b, isfun, islb, tol)=checkredundancy(A, b, A, b, isfun, islb, tol, true)
+
+
+"""Remove redundants cuts in polyhedra (A, b)."""
+function clean!{T}(A::AbstractMatrix{T}, b::AbstractVector{T},
+                   isfun::Bool, islb::Bool, tol::Float64)
+    nincumbents = size(A, 1)
+    redundants = checkredundancy(A, b, isfun, islb, tol)
+    if !isempty(redundants) && length(redundants) < nincumbents
+        tokeep = setdiff(collect(1:nincumbents), redundants)
+
+        A = A[tokeep, :]
```

Comment:
This has no effect on the matrices `A` and `b` of `newcuts!`. We also need to update `mycut`.